src/main/java/org/zalando/nakadi/repository/TopicRepository.java

```diff
@@ -38,7 +39,8 @@ public String getPartition() {
         }
     }
 
-    String createTopic(int partitionCount, Long retentionTimeMs) throws TopicCreationException;
+    String createTopic(int partitionCount, Long retentionTimeMs, CleanupPolicy cleanupPolicy)
```

Comment:
I'ld say this interface has a problem. Every time we need to tweak some configuration in the topic creation you need to change this interface. Maybe we could pass an object with all the configs? Then if we want to tweak clean up minimum latency in the future it becomes easier, just add a property to the object and then map it properly in the implementation, but the interface goes unchanged.

But I honestly cannot think about many other configuration options.

Reply:
Just thought about one: max message size, for example.

Reply:
But I guess we could also do this refactoring when editing it the next time.

Reply:
I think it's not that bad yet. We have some places with 15 parameters :)
I think we should start this "Introduce Parameter Object" refactoring from the most problematic places we have.